python_modules/dagster-graphql/dagster_graphql/implementation/execution/backfill.py

```diff
@@ -95,17 +95,13 @@ def create_and_launch_partition_backfill(
 
     partitions_by_assets = backfill_params.get("partitionsByAssets")
 
-    check.invariant(
-        (
-            asset_selection is None
-            and backfill_params.get("selector") is None
-            and backfill_params.get("partitionNames") is None
-            if partitions_by_assets
-            else True
-        ),
-        "partitions_by_assets cannot be used together with asset_selection, selector, or"
-        " partitionNames",
-    )
+    if (
+        asset_selection or backfill_params.get("selector") or backfill_params.get("partitionNames")
+    ) and partitions_by_assets:
+        raise DagsterError(
```

Comment:
maybe `DagsterInvariantViolationError` or another subclass if you find one you like